mulog-prometheus/src/com/brunobonacci/mulog/publishers/prometheus/collector.clj

```diff
@@ -107,41 +109,42 @@
 
 
 
-(defmulti record-collection (fn [[metric collection]] (:metric-type metric)))
+(defmulti record-collection (fn [[metric collection]] (:metric/metric-type metric)))
 
 (defmethod record-collection :counter
-  [[{:keys [label-values]} collection :as met-col]]
+  [[{:metric/keys [label-values]} collection :as met-col]]
   (increment (child-with-labels collection label-values))
   met-col)
 
 (defmethod record-collection :gauge
-  [[{:keys [label-values metric-value]} collection :as met-col]]
+  [[{:metric/keys [label-values metric-value]} collection :as met-col]]
   (set-value (child-with-labels collection label-values) metric-value)
   met-col)
 
 (defmethod record-collection :histogram
-  [[{:keys [label-values metric-value]} collection :as met-col]]
+  [[{:metric/keys [label-values metric-value]} collection :as met-col]]
   (observe-value (child-with-labels collection label-values) metric-value)
   met-col)
 
 (defmethod record-collection :summary
-  [[{:keys [label-values metric-value]} collection :as met-col]]
+  [[{:metric/keys [label-values metric-value]} collection :as met-col]]
   (observe-value (child-with-labels collection label-values) metric-value)
   met-col)
 
 
 
 (defn cleanup-metrics
   [metrics]
-  (map (fn [{:keys [metric-type metric-value namespace name description labels buckets] :as m}]
+  {:pre [(s/every #(s/valid? ::ms.metric %))]}
```

Comment:
so a metric name is invalid the process with throw exception and stop recording at all?

Reply:
Currently yes this will follow what the java library would do. But it's an interesting discussion point and we could potentially do something about this.

For reference the java library does the following:
metric name validation: [SimpleCollector.java#L164](https://github.com/prometheus/client_java/blob/c41bef4/simpleclient/src/main/java/io/prometheus/client/SimpleCollector.java#L164) -> [Collector.java#L180](https://github.com/prometheus/client_java/blob/c41be/simpleclient/src/main/java/io/prometheus/client/Collector.java#L180)
metric label validation: [SimpleCollector.java#L170](https://github.com/prometheus/client_java/blob/c41bef4/simpleclient/src/main/java/io/prometheus/client/SimpleCollector.java#L170) -> [Collector.java#L201](https://github.com/prometheus/client_java/blob/c41bef/simpleclient/src/main/java/io/prometheus/client/Collector.java#L201)

Reply:
I would say that this is wrong decision... I rather sanitize the label and keep working than halting completely the processing (like in the `metrics` namespace)

Reply:
Yes great thanks for the discussion. Have removed that check - also cleaned the metric accordingly.

The main reason for this was that I was playing with `clojure.spec` and was going to use it for testing. But as discussed I will be using `s/fdef`